src/Compilers/CSharp/Portable/SymbolDisplay/SymbolDisplayVisitor.Types.cs

```diff
@@ -700,6 +706,67 @@ private void AddTypeKind(INamedTypeSymbol symbol)
             }
         }
 
+        /// <summary>
+        /// Copy of <see cref="SynthesizedRecordClone.FindValidCloneMethod(TypeSymbol, ref HashSet{DiagnosticInfo}?)"/>
+        /// </summary>
+        internal static IMethodSymbol FindValidCloneMethod(ITypeSymbol containingType)
+        {
+            IMethodSymbol candidate = null;
+
+            foreach (var member in containingType.GetMembers(WellKnownMemberNames.CloneMethodName))
+            {
+                if (member is IMethodSymbol
+                {
+                    DeclaredAccessibility: Accessibility.Public,
+                    IsStatic: false,
+                    Parameters: { Length: 0 },
+                    Arity: 0
+                } method)
+                {
+                    if (candidate is object)
+                    {
+                        // An ammbiguity case, can come from metadata, treat as an error for simplicity.
+                        return null;
+                    }
+
+                    candidate = method;
+                }
+            }
+
+            if (candidate is null ||
+                !(containingType.IsSealed || candidate.IsOverride || candidate.IsVirtual || candidate.IsAbstract) ||
+                !IsEqualToOrDerivedFrom(
+                    containingType,
+                    candidate.ReturnType,
+                    SymbolEqualityComparer.IgnoreAll))
+            {
+                return null;
+            }
+
+            return candidate;
+
+            bool IsEqualToOrDerivedFrom(ITypeSymbol one, ITypeSymbol other, SymbolEqualityComparer comparison)
```

Comment:
`static`

Reply:
Can this local function be `static`?

---
In reply to: [461958297](https://github.com/dotnet/roslyn/pull/46338#discussion_r461958297) [](ancestors = 461958297)